backend/controllers/users.controller.js

```diff
@@ -0,0 +1,151 @@
+        );
+
+        const user = await new User({
+            name,
+            email,
+            avatar
+        })
+
+        await user.save()
+        await mail(name, email)
+
+        const watchLater = new WatchLater({
+            user: user._id,
+            watchlater: []
+        })
+
+        const LikedVideos = new likedVideos({
+            user: user._id,
+            likedvideos: []
+        })
+
+        const HistoryVideos = new historyVideos({
+            user: user._id,
+            historyvideos: []
+        })
+
+        const playlist = new Playlist({
+            user: user._id,
+            playlists: [
+                {
+                    name: 'fav',
+                    videos: []
+                }
+            ]
+        })
+
+        await watchLater.save()
+        await LikedVideos.save()
+        await HistoryVideos.save()
+        await playlist.save()
+        res.status(200).json({ success: true, message: 'Registration Successfull ' })
+
+    } catch (err) {
+        res.status(500).json({ success: false, message: 'Server Error' })
+    }
+}
+
+//@route   POST api/users/generateotp
+//@desc    Generate 6-digit OTP
+//@access  Public
+const generatedOTP = async (req, res) => {
+    try {
+        const { email } = req.body;
+        let verifyUser = await User.findOne({ email })
+        if (!verifyUser) {
+            return res.status(400).json({ success: false, message: 'Invalid Email' })
+        }
+        const OTP = Number(generateOTP());
+        await OTPMail(OTP, email);
+        const updateUser = {
+            otp: OTP
+        }
+        verifyUser = _.extend(verifyUser, updateUser)
+        await verifyUser.save()
+        res.status(200).send(verifyUser)
+    } catch (err) {
+        res.status(500).json({ success: false, message: 'Server Error' })
+    }
+}
+
+
+//@route   POST api/users/login
+//@desc    Login
+//@access  Public
+const login = async (req, res) => {
+    try {
+        const { email, otp } = req.body;
+        let verifyUser = await User.findOne({ email })
+        if (!verifyUser) {
+            return res.status(400).json({ success: false, message: 'Invalid Email' })
+        }
+
+        if (otp === verifyUser.otp) {
+            verifyUser = _.extend(verifyUser, { otp: null })
+            await verifyUser.save()
+            const token = generateJWToken(verifyUser._id);
+            return res.status(200).send({ token })
+        }
+
+        verifyUser = _.extend(verifyUser, { otp: null })
+        await verifyUser.save()
+        res.status(400).json({ success: false, message: 'Invalid OTP' })
+
+    } catch (err) {
+        res.status(500).json({ success: false, message: 'Server Error' })
+    }
+}
+
+
+//@route   GET api/users
+//@desc    Get User Details
+//@access  Private
+
+const getUserByID = async (req, res) => {
+    try {
+        const id = req.user;
+        const user = await User.findOne({ _id: id }).select(['-createdAt', '-updatedAt', '-__v'])
```

Comment:
if user is null, send 404.

Reply:
It won't be null here because req.user has a token associated with current user and if it was null then it would be handled in auth.js itself
Thanks : )

Reply:
Acchaaaaaa... Right.. Didn't think that through phele.